cmd/cli/root.go

```diff
@@ -81,8 +89,14 @@ func NewRootCmd() *cobra.Command {
 			ctx.Value(util.SystemManagerKey).(*system.CleanupManager).Cleanup(ctx)
 		},
 	}
-
-	RootCmd.PersistentFlags().String("repo", "", "path to bacalhau repo")
+	// ensure the `repo` key always gets a usable default value, warn if it's not.
+	defaultRepo, err := defaultRepo()
```

Comment:
By ensuring there's _always_ a value, does this mean that devstack will no longer be able to default to a /tmp folder?

Reply:
The devstack uses a [different repo flag](https://github.com/bacalhau-project/bacalhau/blob/main/cmd/cli/devstack/devstack.go#L144) from the rest of the code with a [default value](https://github.com/bacalhau-project/bacalhau/blob/main/cmd/cli/devstack/devstack.go#L60) of `""`. It then uses whatever this value is, falling back to Temp if one isn't specified, to [create a repo](https://github.com/bacalhau-project/bacalhau/blob/main/cmd/cli/devstack/devstack.go#L157-L169). This behavior will continue to work as expect despite this change.